### ISSUE TITLE
Distinguish between cookie blocking on network response vs. network requests

### ISSUE THREAD

**Issue by amedina** [GoogleChromeLabs/ps-analysis-tool#381]
## Feature Request

**Description:**
A cookie can blocked by Chrome when it is being set via a server response, as is the case with 3P cookies if 3P cookies being blocked. In this case the cookie will not be available at all. A cookie can also be accepted by Chrome, but later blocked from being set via certain requests. For example when having a 1P cookie with SameSite=Strict being sent to a subdomain of the top domain. 

**User Story:**
As a developer I need to distinguish scenarios where Chrome accepts a cookie on a network response (e.g. `Set Cookie` header), but that cookie is then blocked on some or all network requests (e.g. `Cookie` header). 

**Acceptance Criteria:**
- [x] Developer can distinguish both scenarios
- [x] Information about the number of requests for which a given cookie was blocked
- [x] Evaluate adding "Blocking direction" column, or keep that information on a per-cookie basis. 
- [ ] Filter cookie table by "Blocking direction"

**Comment by kjnanda** [comment]
@ayushnirwal Could you please look into the issue mentioned below.

As discussed the cookie should be shown with warning sign only if it is blocked in one of the response. However, I have found one cookie where I don't see it blocked under `network tab` but PSAT shows it with warning sign.

---

```
Test Data : 
- Test URL - https://edition.cnn.com/
- Cookie:name - akaalb_me-prod-03102020
- Cookie:domain - medium.ngtv.io	
```

-----

Debug log : 

```
{
    "parsedCookie": {
        "name": "akaalb_me-prod-03102020",
        "value": "1707137150~op=aka_prod:aka-east-prod|~rv=37~m=aka-east-prod:0|~os=77af6be3f5f8005d531cfcb6a9745dc4~id=06da929f99eba7a43c94b35748d9dc5c",
        "domain": ".ngtv.io",
        "path": "/",
        "secure": true,
        "httponly": false,
        "samesite": "none",
        "expires": "2024-02-05T12:45:50.000Z",
        "partitionKey": "",
        "size": 157,
        "priority": "Medium"
    },
    "analytics": {
        "platform": "",
        "category": "Uncategorized",
        "name": "",
        "domain": "",
        "description": "",
        "retention": "",
        "dataController": "",
        "gdprUrl": "",
        "wildcard": ""
    },
    "url": "https://medium.ngtv.io/v2/media/livedbcedb554833b248c3ce8374acd2bbcd3983d7dd/desktop?appId=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuZXR3b3JrIjoiY25uIiwicHJvZHVjdCI6ImNubiIsInBsYXRmb3JtIjoid2ViLXRvcDIiLCJhcHBJZCI6ImNubi1jbm4td2ViLXRvcDItOWowYnI2In0.TbUdtroeG7T1gfSTUfdobssbI8vPsAX6tFEX5KI8hcA&ssaiProfile=5I8NQT75Ti",
    "networkEvents": {
        "requestEvents": [],
        "responseEvents": [
            {
                "type": "CHROME_WEBREQUEST_ON_RESPONSE_STARTED",
                "requestId": "10925",
                "url": "https://medium.ngtv.io/v2/media/livedbcedb554833b248c3ce8374acd2bbcd3983d7dd/desktop?appId=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuZXR3b3JrIjoiY25uIiwicHJvZHVjdCI6ImNubiIsInBsYXRmb3JtIjoid2ViLXRvcDIiLCJhcHBJZCI6ImNubi1jbm4td2ViLXRvcDItOWowYnI2In0.TbUdtroeG7T1gfSTUfdobssbI8vPsAX6tFEX5KI8hcA&ssaiProfile=5I8NQT75Ti",
                "blocked": null,
                "timeStamp": 1707137090882
            }
        ]
    },
    "blockingStatus": {
        "inboundBlock": null,
        "outboundBlock": null
    },
    "headerType": "response",
    "isFirstParty": false,
    "frameIdList": [
        0
    ]
}
```

----

Please refer to the screenshots below.

| Tab | Screenshot|
|---- |---- |
| Network Tab | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/d666d262-4018-46a0-b34c-8094897d507f) |
| PSAT | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/c832b404-7134-4fcb-948c-8e0a1096290a)|

**Comment by ayushnirwal** [comment]
@kjnanda Thanks for taking a look at this.

Unfortunately, the correct blocking state is not derivable without the CDP event [Network.responseReceivedExtraInfo](https://chromedevtools.github.io/devtools-protocol/tot/Network/#event-responseReceivedExtraInfo) Which consistently does not fire for this cookie.

While I was looking at the issue I noticed that the Application Panel also misses this cookie, maybe because of the same reason.

For now, for this case where we don't have the correct history in the `networkEvents` property, I will be adding the statement "We could not identify whether this cookie was blocked or not. Please take a look at the network panel by copying the filter string from the cookie's right-click context menu." So a User relies on the network panel for this edge case.

**Comment by maitreyie-chavan** [comment]
Addressed and released as a part of [v0.5](https://github.com/GoogleChromeLabs/ps-analysis-tool/releases/tag/v0.5.0)
